### PR TITLE
Attribute types cannot be decimal

### PR DESCRIPTION
**Pull request by mark-edgeworth**
# Schema Change

This pull request contributes a change to an Open-CMSIS-Pack file schema.
Schema changes have a major impact on the tool ecosystem and need to be done carefully.

## Description

The Packs schema defines two types:`DeviceFeatureType` and `BoardFeatureType`. Both define two attributes `m` and `n`. These attributes were originally assigned a type of xsd:decimal, but this is too restrictive to encompass the range of values permitted for these attributes.

For example: Representation of display resolutions, e.g. `250 x 250`
CMSIS representation: `250.250`. This (originally) parses as a float which is `250.25`, which ends up reading as `250 x 25`

## Checklist
_checklist version 2021-11-10_

- [x] Does this change implement an approved ADR?
       Minor change only, described in the section above.
- [X] Is the schema file changed, properly?
  - [X] Is the schema version incremented according to [Semantic Versioning](https://semver.org/)?
  - [X] Is the schema file header updated, version and date?
  - [X] Is the schema file change history updated?
  - [X] Does the updated schema file pass [XSD meta schema](https://www.w3.org/2012/04/XMLSchema.xsd) check?
- [x] Is the documentation updated, accordingly?
     No update required. The type is only described here, and is implicit in the existing documentation.
- [x] Are the [automatic checks](../tree/main/.github/workflows) updated or enhanced if required?
      Not required. Minor change that does not affect the workflow
- [x] Are enhancements to PackChk required to proof semantic usage of changed/added features?
      No. This is a widening of an existing type to be more lenient.

### CODE REVIEW

**Comment by mark-edgeworth** [comment]
> In order to retain the validation aspect of the schema, I think we should rather define a restricted string type, or am I missing something?
> 
> UPDATED!
> 
> ```
>     <xs:simpleType name="DecimalNumberStringType">
>         <xs:restriction base="xs:string">
>             <xs:pattern value="[ ]*[+-]?(([0-9])|([1-9][0-9]+)|([0-9]\.[0-9]*)|([1-9][0-9]+\.[0-9]*))" />
>         </xs:restriction>
>    </xs:simpleType>
> ```
On reflection I think this is probably only adding complexity to an already-overloaded attribute. Initially the interpretation of the `m` and `n` attributes appears to have been clear (ie as an `xsd:float`) but this has now changed to be something that looks like a float but under some conditions is interpreted as two separate integers separated by a `.`. My concern here is that the various semantics of this field can no longer be fully expressed by XML schema rules (the interpretation depend on the values in other fields). The only way this can be fully validated is by some form of code-based validation that knows the CMSIS interpretation rules. 
In particular, the given regex allows a `+` or `-` at the start of the first integer value, but not the second. Why allow spaces at the start of the number?
So, why not keep it simple and use a simple `xsd:string` here. Handle the semantics when the data is validated (ie on pack creation/submission)?

**Comment by mark-edgeworth** [comment]
@jkrech any comment on this?

**Comment by jkrech** [comment]
Hi @mark-edgeworth, you have listed a number of valid points above and got me thinking what may be the best way forward. I believe we could leave the type xs:decimal in the schema for validation and when reading  you could use the type xs:string to retain the 0. Isn't this the only problem you have detected so far?

**Comment by mark-edgeworth** [comment]
Unfortunately, anything that uses schema generation fails if the declared data type does not match the actual data. This is how we found this issue in the first place. This is not the only problem that we have found, just the first (consider it a pipe-cleaner for the process of changing the schema).

**Comment by mark-edgeworth** [comment]
The checklist items have now been addressed.

**Review by jkrech — APPROVED** [review]
Looks good to me. An open action remains to review whether to enhance PackChk to semantically check that feature descriptions are "valid".

File: schema/PACK.xsd [hunk 1]
```diff
@@ -17,13 +17,16 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 
-  $Date:        1. Dec 2020
+  $Date:        12. Jan 2022
```
Comment (JonatanAntoni) [inline schema/PACK.xsd:20]: $Revision shall reflect latest schema version.

**Comment by JonatanAntoni** [comment]
@mark-edgeworth, can you please rebase this PR. Please take care about final schema version and date.

**Comment by mark-edgeworth** [comment]
@JonatanAntoni can this be merged now please?

**Pull request merged**